Mathlib/MeasureTheory/Integral/IntervalIntegral/IntegrationByParts.lean

```diff
@@ -302,19 +302,19 @@ Compared to `intervalIntegral.integral_comp_smul_deriv` we only require that `g`
 
 If the function `f` is monotone or antitone, see also
 `integral_image_eq_integral_deriv_smul_of_monotoneOn` dropping all assumptions on `g`. -/
-theorem integral_comp_smul_deriv' (h : ∀ x ∈ uIcc a b, HasDerivAt f (f' x) x)
+theorem integral_comp_smul_deriv' (h : ∀ x ∈ uIcc a b, HasDerivWithinAt f (f' x) (uIcc a b) x)
```

Comment:
I think we should have both versions: the one with `HasDerivWithinAt` because it is stronger, and the one with `HasDerivAt` because in the most common situation where the function is differentiable everywhere it will be easier to apply. Maybe name them `integral_comp_smul_derivWithin'` and `integral_comp_smul_deriv'`?

Same thing below.

This should make it possible to revert some changes which don't look so nice in the PR in the applications of these theorems.

Reply:
I mean, keep the current names for the current versions (with the primes and everything) using `HasDerivAt`, and also introduce the versions with `HasDerivWithinAt`, with a name replacing `deriv` with `derivWithin`.